File: Copilot/Java/Factory_Method/round3/source/Cafe.java
public class Cafe {
    private MenuFactory beverageFactory;
    private MenuFactory pastryFactory;
    private MenuFactory snackFactory;

    public Cafe() {
        this.beverageFactory = new BeverageFactory();
        this.pastryFactory = new PastryFactory();
        this.snackFactory = new SnackFactory();
    }

    public MenuItem orderItem(String category, String type) {
        switch (category) {
            case "Beverage":
                return beverageFactory.createMenuItem(type);
            case "Pastry":
                return pastryFactory.createMenuItem(type);
            case "Snack":
                return snackFactory.createMenuItem(type);
            default:
                throw new IllegalArgumentException("Unknown category");
        }
    }
}

File: Copilot/Python/Factory_Method/round1/source/cafe.py
from menu_factory import BeverageFactory, PastryFactory, SnackFactory

class Cafe:
    def __init__(self):
        self.beverage_factory = BeverageFactory()
        self.pastry_factory = PastryFactory()
        self.snack_factory = SnackFactory()

    def order_item(self, category, type):
        if category == "Beverage":
            return self.beverage_factory.create_menu_item(type)
        elif category == "Pastry":
            return self.pastry_factory.create_menu_item(type)
        elif category == "Snack":
            return self.snack_factory.create_menu_item(type)
        else:
            raise ValueError("Unknown category")